codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsProtocolUtils.java

```diff
@@ -98,4 +100,24 @@ static void generateJsonParseBody(GenerationContext context) {
 
         writer.write("");
     }
+
+    /**
+     * Writes an attribute containing information about a Shape's optionally specified
+     * XML namespace configuration to an attribute of the passed node name.
+     *
+     * @param context The generation context.
+     * @param shape The shape to apply the namespace attribute to, if present on it.
+     * @param nodeName The node to apply the namespace attribute to.
+     */
+    static void writeXmlNamespace(GenerationContext context, Shape shape, String nodeName) {
+        shape.getTrait(XmlNamespaceTrait.class).ifPresent(trait -> {
```

Comment:
The XML namespace is inherited from the service too. I don't think this is taking that into account.

Reply:
Setting an XML namespace from the service shape now looks forward to see if the payload structure would have its own namespace, enforcing this inheritance.